internal/scheduler/scheduling_algo.go

```diff
@@ -202,13 +220,46 @@ func (l *LegacySchedulingAlgo) scheduleOnExecutor(
 	updatedJobs := make([]*jobdb.Job, len(jobs))
 	for i, report := range legacyScheduler.SchedulingRoundReport.SuccessfulJobSchedulingReports() {
 		job := report.Job.(*jobdb.Job)
-		job = job.WithQueued(false).WithNewRun(executor.Id)
+		nodeName := ""
+		if len(report.PodSchedulingReports) > 0 {
+			nodeName = report.PodSchedulingReports[0].Node.Name
+		} else {
+			log.Warnf("Could not resolve node for Job %s as no PodSchedulingReports were present", job.Id())
```

Comment:
Should we still lease the job here? Instead of just skip the job

My concern is we'll be leasing it with nodeName as empty
 - Does this not cause issues?

Reply:
I think the right thing to do is to return an error here, which will have the effect of cancelling the entire leasing cycle. I think this is correct because:
- If the legacyScheduler has "leased" the job but hasn't assigned a node then something has gone very wrong and the rest of the leases probably can't be trusted.
- Simply not leasing that job might cause issues- e.g. if that job was part of a gang.

I think we basically need to confirm that we can resolve a valid `nodeName`  and return error if we can't.  This could be because `report.PodSchedulingReports` is empty (already checked) or because `report.PodSchedulingReports` is valid but `report.PodSchedulingReports[0].Node.Name == ""` (not currently checked).